### PR TITLE
Added github actions support for jdwp project

### PR DESCRIPTION
**Pull request by Daquiver1**
This PR tackles issue #55. It creates the GitHub actions workflow for the JDWP projects.

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -3,65 +3,85 @@ name: Build bpftools archives
...
-    - name: Setup ndk
-      run: ./scripts/download-ndk.sh
-    - name: Build
-      run: make bpftrace NDK_PATH=`pwd`/android-ndk-r23b NDK_ARCH=${{ matrix.env['ARCH'] }} STATIC_LINKING=true LLVM_BPF_ONLY=true
-    - uses: actions/upload-artifact@v3
-      with:
-        name: bpftrace-android-${{ matrix.env['ARCH'] }}
-        path: out/android/${{ matrix.env['arch'] }}/bin/bpftrace
+    - name: Navigate to Parent Directory
+      run: cd ..
```
Comment (michalgr) [inline .github/workflows/builds.yml:13]: Why is this necessary ? What directory does execution start in ?

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -3,65 +3,85 @@ name: Build bpftools archives
...
-      run: ./scripts/download-ndk.sh
-    - name: Build
-      run: make bpftrace NDK_PATH=`pwd`/android-ndk-r23b NDK_ARCH=${{ matrix.env['ARCH'] }} STATIC_LINKING=true LLVM_BPF_ONLY=true
-    - uses: actions/upload-artifact@v3
-      with:
-        name: bpftrace-android-${{ matrix.env['ARCH'] }}
-        path: out/android/${{ matrix.env['arch'] }}/bin/bpftrace
+    - name: Navigate to Parent Directory
+      run: cd ..  
+    - name: Initialize submodule folder
```
Comment (michalgr) [inline .github/workflows/builds.yml:14]: Perhaps we could combine this with the first step ?
```
steps:
- name: Checkout repository and submodules
  uses: actions/checkout@v3
  with:
    submodules: recursive
```

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -3,65 +3,85 @@ name: Build bpftools archives
...
+      run: git submodule init && git submodule update
+    - name: Initialize JDWP Project Dependencies
+      run: make buck2-host python-host pyre-host
+    - name: Setup path environment
+      run: eval `make setup-env`
+    - name: Run Buck2 Build
+      run: out/host/bin/buck2 run //projects/jdwp:main
```
Comment (michalgr) [inline .github/workflows/builds.yml:21]: After the command above (`eval $(make setup-env)`) completes you should be able to just call buck2 without the out/host/bin prefix. If that does not work then perhaps each step runs in its own environment, in which case you should be able to do this:
```
  - name: Build and run
    run: |
      eval `make setup-env`
      buck2 run //projects/jdwp:main
  - name: Run tests
    run: |
      eval `make setup-env`
      buck2 test //projects/jdwp/tests/...
  - name: Check types
    run: |
      eval `make setup-env`
      pyre check
```

File: .github/workflows/builds.yml [hunk 4]
```diff
@@ -3,65 +3,85 @@ name: Build bpftools archives
...
+    - name: Run Buck2 Tests
+      run: out/host/bin/buck2 test //projects/jdwp/tests/...
+    - name: Run Pyre Type Checking
+      run: cd projects/jdwp && out/host/bin/pyre
```
Comment (michalgr) [inline .github/workflows/builds.yml:25]: This might not work because after entreing `projects/jdwp` `out/bin/pyre` refers to `ExtendedAndroidTools/projects/jdwp/out/bin/pyre` instead of `ExtendedAndroidTools/out/bin/pyre`

**Review by michalgr — CHANGES_REQUESTED** [review]
In the final version of this pr could you rename current `builds.yml` to `bpftools.yml` and put new logic in `jdwp.yml` ?

File: .github/workflows/builds.yml (hunk 4 above)
Comment (Daquiver1) [reply .github/workflows/builds.yml]: Yeah. Didn't see that. My bad.

**Comment by Daquiver1** [comment]
> LGTM, for the future it would be best to use `git mv` to rename files rather than removing them and adding back with a new name. `git mv` makes it easier to track the history of changes.

Sure, will do.
